driver/sql/projection_notification_processor.go

```diff
@@ -95,37 +104,39 @@ func (b *projectionNotificationProcessor) Start(ctx context.Context, handler Pro
 	return func() {
 		close(b.done)
 		wg.Wait()
-		close(b.queue)
+		b.notificationQueue.Close()
 	}
 }
 
 // Queue puts the notification on the queue to be processed
-func (b *projectionNotificationProcessor) Queue(ctx context.Context, notification *ProjectionNotification) error {
-	select {
-	default:
-	case <-ctx.Done():
-		return context.Canceled
-	case <-b.done:
-		return errors.New("goengine: unable to queue notification because the processor was stopped")
-	}
-
-	b.metrics.QueueNotification(notification)
-
-	b.queue <- notification
-	return nil
+func (b *ProjectionNotificationProcessor) Queue(ctx context.Context, notification *ProjectionNotification) error {
+	return b.notificationQueue.Queue(ctx, notification)
 }
 
-func (b *projectionNotificationProcessor) startProcessor(ctx context.Context, handler ProcessHandler) {
+func (b *ProjectionNotificationProcessor) startProcessor(ctx context.Context, handler ProcessHandler) {
+ProcessorLoop:
 	for {
 		select {
 		case <-b.done:
 			return
 		case <-ctx.Done():
 			return
-		case notification := <-b.queue:
+		case notification := <-b.notificationQueue.Channel():
+			var queueFunc ProjectionTrigger
+			if notification == nil {
+				queueFunc = b.notificationQueue.Queue
+			} else {
+				queueFunc = b.notificationQueue.ReQueue
+
+				if notification.ValidAfter.After(time.Now()) {
```

Comment:
I really wish we could move this into the `Channel()` func what do you think? Would it be possible?